ara_enumerables.rb

```diff
@@ -0,0 +1,129 @@
+# my enumerable
+require './utils.rb'
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    if block_given?
+      arr = []
+      my_each do |i|
+        arr << i if yield(i)
+      end
+    end
+    arr
+  end
+
+  def my_all?(arg = nil)
```

Comment:
* [ ] should have working my_all? identical and returns the same thing as ruby's all?.  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
let us fix this
```p [1, 2i, 3.14].my_all?(Numeric)  != p [1, 2i, 3.14].all?(Numeric)```